agent/directory.go

```diff
@@ -94,10 +96,19 @@ func tryCleanupDirectory(dir string) {
 	}
 
 	_ = filepath.Walk(dir, func(path string, info os.FileInfo, err error) error {
+		if err != nil {
```

Comment:
I think this may be set to nil if finding the file/directory is okay. It's probably better to check for .git before running filepath.Walk. There is a still a risk though. Suppose a user starts an agent in their home directory? I don't know how to avoid that. Maybe check for files/directories starting with a dot in the root, and only then filepath.Walk?

Reply:
we'd need two walks. 

could also (perhaps more easily) see if $HOME is a prefix of dir earlier on?

Reply:
We could check two things:
1. There is not a directory called .git in the directory we want to clean up (before we walk).
2. $HOME is not a prefix of the dir we are going to walk.